neuraxle/base.py

```diff
@@ -151,6 +151,7 @@ def __init__(
             name = self.__class__.__name__
 
         self.name: str = name
+        self.train = train
```

Comment:
I would have declared `self.train = True` just before the last line of the constructor, like just above or just below the `self.is_initialized = False` line since it's related to the lifecycle of the object.

Reply:
Also rename `train` to `is_train` or something like that, and add type: `self.is_train: bool = True`.